java/search-for-webmap/src/main/java/com/esri/arcgisruntime/sample/searchforwebmap/MainActivity.java

```diff
@@ -0,0 +1,231 @@
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // get a reference to the map view
+    mMapView = findViewById(R.id.mapView);
+
+    // load a portal using arcgis.com
+    mPortal = new Portal(getString(R.string.arcgis_url));
+    mPortal.loadAsync();
+
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    getSupportActionBar().setHomeButtonEnabled(true);
+
+    // get a reference to the drawer
+    mDrawer = findViewById(R.id.drawer);
+    mDrawerToggle = new ActionBarDrawerToggle(this, mDrawer, R.string.app_name, R.string.app_name);
+    mDrawerToggle.setDrawerIndicatorEnabled(true);
+    mDrawer.addDrawerListener(mDrawerToggle);
+
+    // setup recycler view
+    mRecyclerView = findViewById(R.id.webmapRecyclerView);
+    mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+    // on reaching the bottom
+    mRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+      @Override
+      public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+        super.onScrollStateChanged(recyclerView, newState);
+        if (!recyclerView.canScrollVertically(1)) {
+          // only get more results if some results have already been returned and no more results are currently being
+          // returned
+          if (mPortalQueryResultSet != null && mMoreResults.isDone()) {
+            getMoreResults();
+          }
+        }
+      }
+    });
+  }
+
+  /**
+   * Searches a portal for webmaps matching query string in the keyword text field. The recycler view is updated with
+   * the results.
+   */
+  private void search(String keyword) {
+    // create query parameters specifying the type webmap
+    PortalQueryParameters params = new PortalQueryParameters();
+    params.setQuery(PortalItem.Type.WEBMAP, null, keyword);
+    // find matching portal items. This search may field a large number of results (limited to 10 be default). Set the
+    // results limit field on the query parameters to change the default amount.
+    ListenableFuture<PortalQueryResultSet<PortalItem>> results = mPortal.findItemsAsync(params);
+    results.addDoneListener(() -> {
+      try {
+        // update the results list view with matching items
+        mPortalQueryResultSet = results.get();
+        mPortalItemList = mPortalQueryResultSet.getResults();
+        PortalItemAdapter portalItemAdapter = new PortalItemAdapter(mPortalItemList,
+            portalItem -> addMap(mPortal, portalItem.getItemId()));
+        mRecyclerView.setAdapter(portalItemAdapter);
+        // open the drawer once there are results
+        mDrawer.openDrawer(Gravity.START);
+        // get 10 more results to fill the recycler view
+        getMoreResults();
+      } catch (Exception e) {
+        String error = "Error getting portal query result set: " + e;
```

Comment:
Probably we can show the custom error only in Toast and not the actual exception
" java.util.concurrent.ExecutionException: java.io.IOException: Loading of dependent object failed" 
as we are already logging it for logcat..

Reply:
Oh good catch thanks. I've changed it to `e.getMessage()` which should now just report "Loading of dependent object failed"